app/shared/components/dynamic-content/tools.ts

```diff
@@ -0,0 +1,17 @@
+import {ComponentInstruction, Router} from 'angular2/router';
+import {appInjector} from '../../tools/app-injector.tool';
+import {PageStructure} from '../../services/page-structure.service';
+
+export function checkContentType(next: ComponentInstruction) {
+  let injector = appInjector();
+  let pageStructure: PageStructure = injector.get(PageStructure);
+  let router: Router = injector.get(Router);
+  /* tslint:disable:no-string-literal */
+  if (pageStructure.hasSectionType(next.params['contentType'])) {
```

Comment:
next.params['contentType']
I guess it can be rewritten as next.params.contentType, in this case we shouldn't suppress linter